src/smtlib_reader.cpp

```diff
@@ -0,0 +1,278 @@
+/*********************                                                        */
+/*! \file smtlib_reader.cpp
+** \verbatim
+** Top contributors (to current version):
+**   Makai Mann
+** This file is part of the smt-switch project.
+** Copyright (c) 2020 by the authors listed in the file AUTHORS
+** in the top-level source directory) and their institutional affiliations.
+** All rights reserved.  See the file LICENSE in the top-level source
+** directory for licensing information.\endverbatim
+**
+** \brief Driver for reading SMT-LIB files. Depends on flex/bison
+**
+**
+**/
+
+#include "smtlib_reader.h"
+
+#include "assert.h"
+#include "smtlibparser.h"
+#include "smtlibparser_maps.h"
+
+using namespace std;
+
+namespace smt {
+
+SmtLibReader::SmtLibReader(smt::SmtSolver & solver)
+    : solver_(solver),
+      def_arg_prefix_("__defvar_")
+{
+  // dedicated true/false symbols
+  // done this way because true/false can be used in other places
+  // for example, when setting options
+  assert(!global_symbols_.current_scope());
+  global_symbols_.add_mapping("true", solver_->make_term(true));
+  global_symbols_.add_mapping("false", solver_->make_term(false));
+}
+
+int SmtLibReader::parse(const std::string & f)
+{
+  file = f;
+  location_.initialize(&file);
+  scan_begin();
+  yy::parser parse(*this);
+  // commented from calc++ example
+  // parse.set_debug_level (trace_parsing);
+  int res = parse();
+  scan_end();
+  return res;
+}
+
+void SmtLibReader::set_logic(const string & logic)
+{
+  solver_->set_logic(logic);
+}
+
+void SmtLibReader::set_opt(const string & key, const string & val)
+{
+  solver_->set_opt(key, val);
+}
+
+void SmtLibReader::set_info(const string & key, const string & val)
+{
+  // No-Op for set-info by default
+}
+
+void SmtLibReader::assert_formula(const Term & assertion)
+{
+  solver_->assert_formula(assertion);
+}
+
+Result SmtLibReader::check_sat()
+{
+  Result r = solver_->check_sat();
+  cout << r << endl;
+  return r;
+}
+
+Result SmtLibReader::check_sat_assuming(const TermVec & assumptions)
+{
+  Result r = solver_->check_sat_assuming(assumptions);
+  cout << r << endl;
```

Comment:
Good question. So for the binary we need it somewhere to print the result. But does it belong in this class or not... I'm not sure. What do you think? I don't want to put it in the actual parser. I chose to put it here so that if you don't want it to print, you can just override the function. But another option is to _not_ print in this class and add it in a derived class.

Reply:
I think it's fine either way.